docs/user/using-an-OIDC-provider.md

```diff
@@ -49,7 +49,86 @@ For Dex, you can find the parameters that you need to set in the configuration (
+
+First, you need to verify that you have `AKS-managed Azure Active Directory: Enabled`.
+
+![Enabled AKS-managed Azure Active Directory](../img/azure-00.png)
+
+Then, you have to register an application. Go to `App registrations and click on `Register an application. Enter a `Name` (any name is ok) and a `Redirect URI` (it should be `https://<your_kubeapps_domain/>oauth2/callback` unless you have manually changed it).
+
+![Registering an application](../img/azure-01.png)
+
+Once you got your application created, click `Overview` to see the `Application (client) ID` and the `Directory (tenant) ID`. You will need these values later.
+
+![Application overview](../img/azure-02.png)
+
+The next step is to create a secret for the application. Click on `Certificates & secrets` and then on `New client secret`.
+
+![Creating secret screen](../img/azure-03.png)
+
+Fill the `Description` and `Expires` fields according to your preferences.
+
+![Creating the secret](../img/azure-04.png)
+
+Next, the `Value` of the secret will appear and you will be able to copy it. You will need this value later.
+
+![Retrieving the secret value](../img/azure-05.png)
+
+The following step is to define the permissions that the application will need. To do so, go to `API permissions` and click `Add a permission`.
+
+![Add permissions screen](../img/azure-06.png)
+
+Then, in the `APIs my organization uses` enter this value `6dae42f8-4368-4678-94ff-3960e28e3630` (this is the ID corresponding to the AKS AAD server).
+
+![Selecting the permissions](../img/azure-07.png)
+
+Next, select the `user.read` permission and click `Add permissions`.
+
+![Selecting user.read permission](../img/azure-08.png)
+
+Then, go to `Manifest` and change `accessTokenAcceptedVersion` from `null` to `2`. It will make the application generate v2 tokens instead of v1.
+
+![Editing the manifest](../img/azure-09.png)
+
+At the end of these steps, you will have created an application, granted it with `user.read` permissions and changed the token version to v2. You also will have this information.
+
+- `Application (client) ID`, for instance, `my-application-id`.
+- `Application secret`, for instance, `my-secret`.
+- `Directory (tenant) ID`, for instance, `my-tenant-id`.
+
+The next step is just configuring Kubeapps to use all these values. We highlight here:
+
+- The issuer is `https://login.microsoftonline.com/my-tenant/v2.0` (replacing `my-tenant` with your own value), since we are using v2 tokens.
+- The scope is exactly `openid 6dae42f8-4368-4678-94ff-3960e28e3630/user.read`, this way, you will able to get the user's email as well as get access to the protected resource that is AKS.
+
+> In v1 tokens, you had to pass this value as part of the `--resource=6dae42f8-4368-4678-94ff-3960e28e3630` flag, but in v2 tokens, this claim is performed just using the scope.
+
+```yaml
+frontend:
+  proxypassAccessTokenAsBearer: true # required to pass the access_token instead of the id_token
```

Comment:
Ah - excellent, so that was related.

Reply:
Yep, I mean, it wasn't the root cause of all my pain since I was just copying the token from the response (exposed with `--set-xauthrequest=true` as this person recommended) and using it back in postman (to avoid extra layers).
The issue was that I always got  401 errors... (or even if the oauth2proxy container crashed while using the azure provider...)